aeron-driver/src/main/java/io/aeron/driver/Configuration.java

```diff
@@ -1096,4 +1098,68 @@ public static void validatePageSize(final int pageSize)
             throw new ConfigurationException("Page size not a power of 2: page size=" + pageSize);
         }
     }
+
+    private static int getSize(final String propertyKey, final int defaultValue)
+    {
+        final String propString = getProperty(propertyKey);
+        if (propString != null)
+        {
+            return parseSize(propertyKey, propString);
+        }
+        else
+        {
+            return defaultValue;
+        }
+    }
+
+    private static long getSize(final String propertyKey, final long defaultValue)
+    {
+        final String propertyValue = getProperty(propertyKey);
+        if (propertyValue != null)
+        {
+            return parseSize(propertyKey, propertyValue);
+        }
+        else
+        {
+            return defaultValue;
+        }
+    }
+
+    private static final Pattern SIZE_REGEX = Pattern.compile("(?<value>\\d+)(?<magnitude>\\w)?");
+    private static final String PARSE_SIZE_MAGNITUDE_FAILURE_FORMAT = "Couldn't parse value: %s for property %s. " +
+            "Trailing character should be one of k, m, or g, but was %s";
+    private static final String PARSE_SIZE_VALUE_FAILURE_FORMAT = "Couldn't parse value: %s for property %s. " +
+            "Should be a positive number followed by one of k, m, or g.";
+
+    public static int parseSize(final String propertyKey, final String sizeString)
+    {
+        final Matcher matcher = SIZE_REGEX.matcher(sizeString);
+        if (!matcher.matches())
+        {
+            throw new IllegalArgumentException(String.format(PARSE_SIZE_VALUE_FAILURE_FORMAT, sizeString, propertyKey));
+        }
+
+        final int value = Integer.valueOf(matcher.group("value"));
+        final String magnitude = matcher.group("magnitude");
+
+        if (magnitude != null)
+        {
+            switch (magnitude)
+            {
+                case "k":
+                    return value * 1024;
+                case "m":
+                    return value * 1024 * 1024;
+                case "g":
+                    return value * 1024 * 1024 * 1024;
+                default:
+                    throw new IllegalArgumentException(
+                            String.format(PARSE_SIZE_MAGNITUDE_FAILURE_FORMAT, sizeString, propertyKey, magnitude));
+            }
```

Comment:
Profiling has shown us that String.format is very costly. I appreciate that this is an exception case but in general we are trying to avoid its use.